backend/internal/app/app.go

```diff
@@ -0,0 +1,102 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"shodo/internal/config"
+	"shodo/internal/domain/services"
+	"shodo/internal/repository"
+	"shodo/internal/transport"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-redis/redis"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.uber.org/zap"
+)
+
+func Run(r *gin.Engine) {
+
+	Logger, err := zap.NewProduction()
+	if err != nil {
+		panic(err)
+	}
+
+	configName := os.Getenv("CONFIG_NAME")
+	config, err := config.Init(configName)
+	if err != nil {
+		fmt.Println(err)
+		panic(err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	mongoUrl := fmt.Sprintf("mongodb://admin:password@%s:27017", config.MongoHost)
```

Comment:
Best practice is to move this to config. You already have a MongoHost in config, why not credentials?
You can even go one step further and encode password, but that might not be necessary

Reply:
Moved